methods/dst.py

```diff
@@ -116,31 +119,34 @@ async def search_images():
         with TimeIt("image comparisons"):
             out_dir = os.path.join("compare_screens", url_hash)
 
-            # TODO: implement concurrency
+            screenshot_group: FutureGroup = screenshot_worker.new_future_group()
+
             # Check all found URLs
             for index, resulturl in enumerate(url_list_text + url_list_img):
-                if check_image(out_dir, index, session_file_path, resulturl):
-                    # Match for found images, so conclude as phishing
-                    logger.info(f"[RESULT] Phishing, for url {url}, due to image comparisons with index {index}: {resulturl}")
+                screenshot_group.schedule([out_dir, index, session_file_path, resulturl])
 
-                    return ResultType.PHISHING
+            if screenshot_group.any(): # Match for found images, so conclude as phishing
+                logger.info(f"[RESULT] Phishing, for url {url}, due to image comparisons with index {index}: {resulturl}")
+                screenshot_group.cancel()
+                return ResultType.PHISHING
 
         # If the inconclusive stems from google blocking:
         #   e.g. blocked == True
         #   result: inconclusive_blocked
 
         logger.info(f"[RESULT] Inconclusive, for url {url}")
 
+        worker.close()
```

Comment:
why do we close it here? we should be able to reuse it for multiple detections, cause it's not going to be reopened after but another run on the same dst object will use it